### PR TITLE
reorg repo structure to prepare for ee features

### PR DESCRIPTION
**Pull request by itayd**
this puts all of backend into internal, exposing only a small surface of it to be used for embedding. this will allow to add an `ee` (temporary name) folder for enterprise features that have access to all the inards of the backend.

### CODE REVIEW

File: tests/system/testdata/manifest/apply.txtar [hunk 1]
```diff
@@ -16,12 +16,12 @@ output equals ''
 # Successful dry run of a full project.
 ak manifest apply full.yaml --dry-run -J
 return code == 0
-output equals file expected_on_full_dry_run.json
+output equals file expected_on_full_dry_run.txt
```
Comment (daabr) [inline tests/system/testdata/manifest/apply.txtar:19]: I agree with changing the extension, but can you also remove the unnecessary `-J` from lines 17 and 22?

File: internal/xdg/xdg.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-Package config manages autokitteh's configuration and data directories,
+This manages autokitteh's configuration and data directories,
```
Comment (daabr) [inline internal/xdg/xdg.go]: Package documentation should begin with "Package <Name>..." - see https://tip.golang.org/doc/comment. Bug given the package name change, it should be "Package XDG manages..."

**Review by daabr — APPROVED** [review]
Approved with 2 tiny nits.

File: tests/system/testdata/manifest/apply.txtar (hunk 1 above)
Comment (itayd) [reply tests/system/testdata/manifest/apply.txtar:19]: done

File: internal/xdg/xdg.go (hunk 2 above)
Comment (itayd) [reply internal/xdg/xdg.go]: done

**Pull request merged**